File: src/main/java/xyz/phanta/cbtweaker/buffer/ingredient/IngredientMatcherType.java
package xyz.phanta.cbtweaker.buffer.ingredient;

import com.google.gson.JsonObject;
import net.minecraft.util.ResourceLocation;

public interface IngredientMatcherType<A, JA> {

    ResourceLocation getId();

    IngredientMatcher<A, JA> loadMatcher(JsonObject config);

}

File: src/main/kotlin/st/evening/mc/cbtweaker/buffer/ingredient/IngredientMatcherType.kt
package st.evening.mc.cbtweaker.buffer.ingredient

import st.evening.mc.cbtweaker.util.Identifiable
import st.evening.mc.prelude.api.data.tjson.JsonPath
import st.evening.mc.prelude.api.data.tjson.TJson

interface IngredientMatcherType<A, JA> : Identifiable {
    context(_: JsonPath)
    fun loadMatcher(dto: TJson.Object): IngredientMatcher<A, JA>
}
